initrd/bin/oem-factory-reset

```diff
@@ -276,43 +278,85 @@ fi
 
 # show warning prompt
 if [ "$CONFIG_TPM" = "y" ]; then
-    TPM_STR="          * ERASE the TPM and reset it with a default password\n"
+    TPM_STR="          * ERASE the TPM and own it with a password\n"
 else
     TPM_STR=""
 fi
 if ! whiptail --yesno "
-        This operation will automatically:\n\n
+        This operation will automatically:\n
 $TPM_STR
           * ERASE any keys or passwords on the GPG smart card,\n
-            reset it to a factory state, and generate new keys\n
+            reset it to a factory state, generate new keys\n
+            and optionally set custom PIN(s)
           * Add the new GPG key to the firmware and reflash it\n
           * Sign all of the files in /boot with the new GPG key\n\n
         It requires that you already have an OS installed on a\n
-        dedicated /boot partition. Do you wish to continue?\n" \
+        dedicated /boot partition. Do you wish to continue?" \
         $HEIGHT $WIDTH $CONTINUE $CANCEL $CLEAR $bg_color --title "$title_text" ; then
     exit 1
 fi
 
 # Prompt to change default passwords
-echo -e -n "Would you like to set a custom password? [y/N]: "
+echo -e -n "Would you like to set a single custom password applied to physical security components? [y/N]: "
 read -n 1 prompt_output
 echo
 if [ "$prompt_output" == "y" \
   -o "$prompt_output" == "Y" ] \
 ; then
-  echo -e "\nThe custom password will be used for the
-TPM admin and GPG user/admin passwords.
+  echo -e "\nThe chosen custom password will be applied to:
```

Comment:
I think you want to inform the user of this *before* they make the selection, not after

Reply:
@mrchromebox : "custom password will be applied to all security components" and the user said yes. The other option is to define different pins/passwords for different components.

 I basically separated the output as it was before to be more clear on what was going to be applied and to be able to add more components having the same password if needed before applying that custom password to all components.

This PR was to be able to apply different passwords to different security components and output provisionned passwords before reboot.

You want the concerned security components to be clearer prior of engaging action?

Reply:
it just seems odd to enumerate what the password will be applied to after the user has made the choice rather than before. I would just move the echo block up before the read call

Reply:
@MrChromebox : better? https://github.com/osresearch/heads/pull/1136#issuecomment-1065250513